policy-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/policy/common/ContractTestData.java

```diff
@@ -0,0 +1,174 @@
+import uk.gov.gchq.palisade.rule.Rule;
+import uk.gov.gchq.palisade.service.ConnectionDetail;
+import uk.gov.gchq.palisade.service.SimpleConnectionDetail;
+import uk.gov.gchq.palisade.service.policy.model.PolicyRequest;
+import uk.gov.gchq.palisade.service.policy.model.Token;
+import uk.gov.gchq.palisade.util.ResourceBuilder;
+
+import java.io.Serializable;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Common test data for all classes
+ * This cements the expected JSON input and output, providing an external contract for the service
+ */
+public class ContractTestData {
+
+    private ContractTestData() {
+        // hide the constructor, this is just a collection of static objects
+    }
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public static class PassThroughRule<T extends Serializable> implements Rule<T> {
+        @Override
+        public T apply(final T record, final User user, final Context context) {
+            return record;
+        }
+    }
+
+    public static final UserId USER_ID = new UserId().id("test-user-id");
+    public static final UserId NO_RESOURCE_RULES_USER_ID = new UserId().id("noResourceRulesUser");
+    public static final String RESOURCE_ID = "/test/resourceId";
+    public static final String NO_RESOURCE_RULES_RESOURCE_ID = "/test/noRulesResource";
+    public static final String PURPOSE = "test-purpose";
+    public static final String NO_RESOURCE_RULES_PURPOSE = "test-purpose";
+    public static final Context CONTEXT = new Context().purpose(PURPOSE);
+    public static final Context NO_RESOURCE_RULES_CONTEXT = new Context().purpose(NO_RESOURCE_RULES_PURPOSE);
+    public static final User USER = new User().userId(USER_ID).roles("role").auths("auth");
+    public static final User NO_RESOURCE_RULES_USER = new User().userId(NO_RESOURCE_RULES_USER_ID).roles("role").auths("auth");
+    public static final ConnectionDetail CONNECTION_DETAIL = new SimpleConnectionDetail().serviceName("test-data-service");
+    public static final LeafResource RESOURCE = (LeafResource) ResourceBuilder.create(RESOURCE_ID);
+    public static final LeafResource NO_RESOURCE_RULES_RESOURCE = (LeafResource) ResourceBuilder.create(NO_RESOURCE_RULES_RESOURCE_ID);
+    public static final PolicyRequest POLICY_REQUEST;
+    public static final PolicyRequest NO_RESOURCE_RULES_POLICY_REQUEST;
+    public static final JsonNode REQUEST_NODE;
+    public static final JsonNode NO_RESOURCE_RULES_REQUEST_NODE;
+    public static final PolicyRequest REQUEST_OBJ;
+    public static final PolicyRequest NO_RESOURCE_RULES_REQUEST_OBJ;
+    public static final String REQUEST_JSON;
+    public static final String NO_RESOURCE_RULES_REQUEST_JSON;
+
+    static {
+        RESOURCE.connectionDetail(CONNECTION_DETAIL).serialisedFormat("txt").setType("test");
+        POLICY_REQUEST = PolicyRequest.Builder.create()
```

Comment:
although the builder is used, the object is marhsalled into JSON, but then not used for the assertion, if the test used the builder to create the object, and then use the builder to assert against the original object then it would be a json seraliser/deseraliser test, but as the test, lines 209-219 in the kafka contract test, are using string equivalents to validate against the object, so dont fall into the risk where you havent had to update the tests when making a change

Reply:
updated